### PR TITLE
PR for edit and filter feature

### PR DESCRIPTION
**Pull request by narharikale**
added edit feat
added priority feat

### CODE REVIEW

**Review by prathmeshjagtap — APPROVED** [review]

File: src/frontend/components/modal/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                      <button className='notsy-btn notesy-btn notesy-secondary-btn ' onClick={ () => setNoteModal(false)}>Close</button>
+                    </div>
+                  </div>
+                </div>
+              </div>
+
+                
+                
+              
+            </div>
```
Comment (horsemaker) [inline src/frontend/components/modal/Modal.jsx:51]: You could use some code formatter to format these white spaces.

File: src/frontend/context/Filter-context/Filter-context.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { createContext, useContext} from "react";
+import { useReducer } from "react";
+
+const FilterContext = createContext();
+
+
+
+function FilterProvider( { children } ) {
```
Comment (saketkum11) [inline src/frontend/context/Filter-context/Filter-context.js:8]: Great way to have component

File: src/frontend/components/noteCard/noteCard.jsx [hunk 3]
```diff
@@ -3,98 +3,144 @@ import "./noteCard.css";
...
-      {priority && <div class="badge-text">{priority}</div>}
-      <div className="note-card-header">
-        <div className="note-title"> {title} </div>
-        {/* <button className="icon-btn material-icons-outlined"> push_pin </button> */}
-      </div>
+      <div className={`note-card ${color} `}>
+        {priority && <div className="badge-priority">{priority}</div>}
+        <div className="note-card-header">
+          <div className="note-title"> {title} </div>
+          {/* <button className="icon-btn material-icons-outlined"> push_pin </button> */}
```
Comment (horsemaker) [inline src/frontend/components/noteCard/noteCard.jsx:63]: You could remove the comments.

**Review by omgaikwad — APPROVED** [review]
Good use of context, clean code, good file structure, you are good to go!

**Review by horsemaker — APPROVED** [review]
Final touches! 🪄

**Review by saketkum11 — APPROVED** [review]
- clean code.
- easy to  understandable.

**Review by t007rushi — APPROVED** [review]
Clean code practices followed 
well structured directory system
I like the Logo seems like bluestack

**Pull request merged**